### PR TITLE
Add description of vshard.enable/disable

### LINKED ISSUE #3012: vshard.router.enable/disable()
**Product:** vshard
**Since:** product version
**Audience/target:** dev, admin
**Root document:** https://www.tarantool.io/en/doc/latest/reference/reference_rock/vshard/vshard_router/
(source: https://github.com/tarantool/doc/blob/latest/doc/reference/reference_rock/vshard/vshard_router.rst)
**SME:** @ Serpentian

# Details

`vshard.router.disable()` makes most of the `vshard.router`
functions throw an error. As Lua exception, not via `nil, err`
pattern.

`vshard.router.enable()` reverts the disable.

`router_object:enable()/disable()`, where `router_object` is
the return value of `vshard.router.new()`, can also be used
for manual API access configuration for the specific non-static
router.

By default the router is enabled.

Additionally, the router is forcefully disabled automatically
until its configuration is finished and the instance finished
recovery (its `box.info.status` is `'running'`, for example).

Auto-disable protects from usage of vshard functions before the
router's global state is fully created.

Manual router's disabling helps to achieve the same for user's
application. For instance, a user might want to do some preparatory
work after `vshard.router.cfg` before the application is ready.
Then the flow would be:
```Lua
vshard.router.disable()
vshard.router.cfg(...)
-- Do your preparatory work here ...
vshard.router.enable()
```

The behavior of the router's API enabling/disabling is similar to the
storage's one.
Requested by @Serpentian in https://github.com/tarantool/vshard/commit/dd70cfb2c5ec36ab7d5355b0024e5f6d21bb8f9f.

### PR DESCRIPTION
**Pull request by Serpentian**
* Add description of the automatic and manual router API's
  enabling/disabling

Resolves #3012

### CODE REVIEW

File: doc/reference/reference_rock/vshard/vshard_router.rst [hunk 1]
```diff
@@ -768,6 +770,46 @@ Router public API
...
+..  function:: vshard.router.enable()
+
+    Manually allow access to the router API, revert
+    :ref:`vshard.router.disable() <router_api-disable>`.
+
+        .. NOTE::
+
+           ``vshard.router.enable()`` cannot be used for enabling the router
+           API which was automatically disabled due to running configuration
+           process.
```
Comment (patiencedaur) [inline doc/reference/reference_rock/vshard/vshard_router.rst:784]:
```suggestion
        ..  note::

            ``vshard.router.enable()`` cannot be used for enabling a router
            API that was automatically disabled due to a running configuration
            process.
```
1. We use `..  note::` now: https://www.tarantool.io/en/doc/latest/contributing/docs/markup/admonitions/
2. A running configuration process is `router.cfg()`, right? Let's mention this.

File: doc/reference/reference_rock/vshard/vshard_router.rst [hunk 2]
```diff
@@ -768,6 +770,46 @@ Router public API
...
+
+..  function:: vshard.router.disable()
+
+    Manually restrict access to the router API. Disabling makes all
+    its methods, excluding :ref:`vshard.router.cfg() <router_api-cfg>`,
+    :ref:`vshard.router.new() <router_api-new>`,
+    :ref:`vshard.router.enable() <router_api-enable>` and
+    :ref:`vshard.router.disable() <router_api-disable>`, to throw a Lua
+    error in case of invoking them. The error object has a name attribute
+    equal to ``ROUTER_IS_DISABLED``.
```
Comment (patiencedaur) [inline doc/reference/reference_rock/vshard/vshard_router.rst:796]:
```suggestion
    Manually restrict access to the router API. When the API is disabled,
    all its methods throw a Lua error,
    except :ref:`vshard.router.cfg() <router_api-cfg>`,
    :ref:`vshard.router.new() <router_api-new>`,
    :ref:`vshard.router.enable() <router_api-enable>` and
    :ref:`vshard.router.disable() <router_api-disable>`. The error object's name attribute is
    ``ROUTER_IS_DISABLED``.
```
Russian is a wordy language, and English is not. Please try to [use fewer words and simpler constructions](https://www.tarantool.io/en/doc/latest/contributing/docs/style/#concise-is-good) whenever possible.

File: doc/reference/reference_rock/vshard/vshard_router.rst [hunk 3]
```diff
@@ -768,6 +770,46 @@ Router public API
...
+
+    The router is enabled by default. However, it is automatically and
+    forcefully disabled until configuration is finished as accessing the
```
Comment (patiencedaur) [inline doc/reference/reference_rock/vshard/vshard_router.rst:799]:
```suggestion
    forcefully disabled until the configuration is finished, as accessing the
```

**Review by patiencedaur — CHANGES_REQUESTED** [review]
Thanks for the patch! Most of my suggestions are language-related.
I haven't looked at the changes on the staging website yet because the deployment is currently broken. If it's okay with you, I'll take another look at the PR after the deployment is fixed.

**Comment by Serpentian** [comment]
Closed as a duplicate of #3025

**Pull request closed without merging**